atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/BackgroundSweeperImpl.java

```diff
@@ -309,13 +314,15 @@ private void saveSweepResults(final SweepProgressRowResult progress,
     private void saveIntermediateSweepResults(final SweepProgressRowResult progress,
                                               final byte[] nextStartRow,
                                               final long cellsDeleted,
-                                              final long cellsExamined) {
+                                              final long cellsExamined,
+                                              final long minimumSweptTimestamp) {
         txManager.runTaskWithRetry(new TxTask() {
             @Override
             public Void execute(Transaction t) {
                 SweepProgressTable progressTable = tableFactory.getSweepProgressTable(t);
                 SweepProgressRow row = SweepProgressRow.of(0);
                 progressTable.putFullTableName(row, progress.getFullTableName());
+                progressTable.putMinimumSweptTimestamp(row, minimumSweptTimestamp);
```

Comment:
I believe it should be written every time and a problem actually occurs because the value isn't read out of the priority table when constructing the new progress object from chooseNextTable.  This means that there could be a case where sweep runs entirely at TS = X final, and the subsequent one runs at an earlier TS, though thinking about this I'm not sure that's entirely a problem considering we may have already truncated up to the later timestamp.  We can discuss the example in person.